packages/utils/src/feature-flags/utils/flag-router.ts

```diff
@@ -19,11 +19,17 @@ export class FlagRouter implements FeatureFlagTypes.IFlagRouter {
    * @param flag - The flag to check
    * @return {boolean} - Whether the flag is enabled or not
    */
-  public isFeatureEnabled(flag: string | Record<string, string>): boolean {
+  public isFeatureEnabled(
+    flag: string | string[] | Record<string, string>
+  ): boolean {
     if (isString(flag)) {
       return !!this.flags[flag]
     }
 
+    if (Array.isArray(flag)) {
+      return flag.every((f) => !!this.flags[f])
+    }
+
     if (isObject(flag)) {
       const [nestedFlag, value] = Object.entries(flag)[0]
```

Comment:
```suggestion
  public isFeatureEnabled(
    flag: string | string[] | Record<string, string>
  ): boolean {
    if (isObject) {
      const [nestedFlag, value] = Object.entries(flag)[0]

      if (typeof this.flags[nestedFlag] === "boolean") {
        return this.flags[nestedFlag] as boolean
      }

      return !!this.flags[nestedFlag]?.[value]
    }
    
    const flags = Array.isArray(flag) ? flag : [flag]
    return flag.every((flag_) => {
      if (!isString(flag_)) {
        throw Error("Flag must be a string an array of string or an object")
      }
      return !!this.flags[f])
    }
```

Reply:
this didn't work?

Reply:
`typeof` array is object so we need to check first for the array case and return early

Reply:
<img width="435" alt="Screenshot 2023-12-21 at 17 12 09" src="https://github.com/medusajs/medusa/assets/25098370/973dbee5-a7ad-43ac-af3c-e77ef3266c82">

Reply:
ah ok the issue here is that another `isObject` util form `utils/src/common` is used in this place ATM...I will refactor

Reply:
It is non blocking though, more of a suggestion 👌

Reply:
I think one think to improve about our isObject utils would be the following:

typeof value === 'object' &&
    value != null &&
    !Array.isArray(value) &&
!isDate(value)

Wdyt?